docs/Parameters-Tuning.rst

```diff
@@ -36,15 +36,143 @@ To get good results using a leaf-wise tree, these are some important parameters:
+
+The suggestions below will speed up training, but might hurt statistical performance.
+
+Enable Feature Pre-Filtering When Creating Dataset
+**************************************************
+
+By default, when a LightGBM ``Dataset`` object is constructed, some features will be filtered out based on the value of ``min_data_in_leaf``.
+
+For a simple example, consider a 1000-observation dataset with a feature called ``feature_1``. ``feature_1`` takes on only two values: 25.0 (995 observations) and 50.0 (5 observations). If ``min_data_in_leaf = 10``, there is no split for this feature which will result in a valid split at least one of the leaf nodes will only have 5 observations.
+
+Instead of reconsidering this feature and then ignoring it every iteration, LightGBM filters this feature out at before training, when the ``Dataset`` is constructed.
+
+If this default behavior has been overridden by setting ``feature_pre_filter=False``, set ``feature_pre_filter=True`` to reduce training time.
+
+Decrease ``max_bin`` or ``max_bin_by_feature`` When Creating Dataset
+********************************************************************
+
+LightGBM training `buckets continuous features into discrete bins <./Features.rst#optimization-in-speed-and-memory-usage>`_ to improve training speed and reduce memory requirements for training. This binning is done one time during ``Dataset`` construction. The number of splits considered when adding a node is ``O(#feature * #bin)``, so reducing the number of bins per feature can reduce the number of splits that need to be evaluated.
+
+``max_bin`` is controls the maximum number of bins that features will bucketed into. It is also possible to set this maximum feature-by-feature, by passing ``max_bin_by_feature``.
+
+Reduce ``max_bin`` or ``max_bin_by_feature`` to reduce training time.
+
+Increase ``min_data_in_bin`` When Creating Dataset
+**************************************************
+
+Some bins might contain a small number of observations, which might mean that the effort of evaluating that bin's boundaries as possible split points isn't likely to change the final model very much. You can control the granularity of the bins by setting ``min_data_in_bin``.
+
+Increase ``min_data_in_bin`` to reduce training time.
+
+Decrease ``feature_fraction``
+*****************************
+
+By default, LightGBM considers all features in a ``Dataset`` during the training process. This behavior can be changed by setting ``feature_fraction`` to a value ``> 0`` and ``<= 1.0``. Setting ``feature_fraction`` to ``0.5``, for example, tells LightGBM to randomly select ``50%`` of features at the beginning of constructing each tree. This reduces the total number of splits that have to be evaluated to add each tree node.
+
+Decrease ``feature_fraction`` to reduce training time.
+
+Decrease ``max_cat_threshold``
```

Comment:
I based this paragraph on the explanations in https://github.com/microsoft/LightGBM/issues/2261, https://github.com/microsoft/LightGBM/issues/2411, and https://github.com/microsoft/LightGBM/issues/2919